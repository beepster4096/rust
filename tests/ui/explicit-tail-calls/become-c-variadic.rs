#![feature(explicit_tail_calls)]
#![feature(c_variadic)]

pub unsafe extern "C" fn c_variadic_f(x: u8, mut args: ...) {
    become c_variadic_g(x)
    //~^ error: tail-calls are not allowed in c-variadic functions
    //~| error: c-variadic functions can't be tail-called
}

pub unsafe extern "C" fn c_variadic_g(x: u8, mut args: ...) {
    become normal(x)
    //~^ error: tail-calls are not allowed in c-variadic functions
}

unsafe extern "C" fn normal(x: u8) {
    become c_variadic_f(x)
    //~^ error: c-variadic functions can't be tail-called
}

fn main() {}
