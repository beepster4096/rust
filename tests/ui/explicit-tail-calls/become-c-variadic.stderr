error: tail-calls are not allowed in c-variadic functions
  --> $DIR/become-c-variadic.rs:5:5
   |
LL |     become c_variadic_g(x)
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: c-variadic functions can't be tail-called
  --> $DIR/become-c-variadic.rs:5:5
   |
LL |     become c_variadic_g(x)
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: tail-calls are not allowed in c-variadic functions
  --> $DIR/become-c-variadic.rs:11:5
   |
LL |     become normal(x)
   |     ^^^^^^^^^^^^^^^^

error: c-variadic functions can't be tail-called
  --> $DIR/become-c-variadic.rs:16:5
   |
LL |     become c_variadic_f(x)
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

